src/main/java/org/fsg1/fmms/backend/endpoints/CurriculaEndpoint.java

```diff
@@ -0,0 +1,49 @@
+package org.fsg1.fmms.backend.endpoints;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.fsg1.fmms.backend.services.CurriculaService;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * The class containing the curricula endpoints.
+ */
+@Path("curricula")
+public class CurriculaEndpoint {
+
+    private final CurriculaService service;
+
+    /**
+     * Constructor which receives the service as dependency.
+     *
+     * @param curriculaService CurriculaService object.
+     */
+    @Inject
+    public CurriculaEndpoint(final CurriculaService curriculaService) {
+        service = curriculaService;
+    }
+
+    /**
+     * Returns all curricula.
+     *
+     * @return A JSON array of all the curricula.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getCurriculumSemesters() {
```

Comment:
Don't you need a `@Path` here?

Reply:
The `@Path` is inherited from the class's `@Path` annotation. 

```
@Path("curricula")
public class CurriculaEndpoint {
```